File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1 for i in range(len(nums))]
        for i in range(1, len(nums)):
            res[i] = res[i-1] * nums[i-1]
        k = 1
        for i in range(len(nums) - 2, -1, -1):
            k *= nums[i+1]
            res[i] *= k
        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num : nums){
            set.add(num);
        }
        int longestStreak = 0;
        for(int num : nums){
            if(set.contains(num - 1)) continue;
            int currentStreak = 1;
            int currentNum = num + 1;
            while(set.contains(currentNum)){
                currentStreak++;
                currentNum++;
            }
            longestStreak = Math.max(longestStreak, currentStreak);
        }
        return longestStreak;
    }
}
